File: leetcode/275. H 指数 II.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        if len(citations) == 1:
            return 1 if citations[0] > 0 else 0
        for i in range(len(citations), 0, -1):
            if citations[len(citations) - i] >= i:
                return i
        return 0

File: leetcode/275. H 指数 II.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        if (citations.size() == 1) {
            return citations[0] >= 1 ? 1 : 0;
        }
        for (int i = citations.size(); i > 0; i--) {
            if (citations[(int)citations.size() - i] >= i) {
                return i;
            }
        }
        return 0;
    }
};